src/reformat_libyuv.c

```diff
@@ -350,429 +368,136 @@ avifResult avifImageYUVToRGBLibYUV8bpc(const avifImage * image,
 
     assert((image->depth == 8) && (rgb->depth == 8));
 
-#if LIBYUV_VERSION >= 1813
-    enum FilterMode filter = kFilterBilinear;
-    if (rgb->chromaUpsampling == AVIF_CHROMA_UPSAMPLING_FASTEST) {
-        // 'None' (Nearest neighbor) filter is faster than bilinear.
-        filter = kFilterNone;
-    }
-#endif
-
-    // This following section might be a bit complicated to audit without a bit of explanation:
-    //
     // libavif uses byte-order when describing pixel formats, such that the R in RGBA is the lowest address,
     // similar to PNG. libyuv orders in word-order, so libavif's RGBA would be referred to in libyuv as ABGR.
     // In addition, swapping U and V in any of these calls, along with using the Yvu matrix instead of Yuv matrix,
-    // swaps B and R in these orderings as well. This table summarizes this block's intent:
+    // swaps B and R in these orderings as well. This table summarizes the look up tables that follow:
     //
     // libavif format            libyuv Func      UV matrix (and UV argument ordering)
     // --------------------      -------------    ------------------------------------
     // AVIF_RGB_FORMAT_RGB       *ToRGB24Matrix   matrixYVU
+    // AVIF_RGB_FORMAT_RGBA      *ToARGBMatrix    matrixYVU
+    // AVIF_RGB_FORMAT_ARGB      *ToRGBAMatrix    matrixYVU
     // AVIF_RGB_FORMAT_BGR       *ToRGB24Matrix   matrixYUV
-    // AVIF_RGB_FORMAT_RGB_565   *ToRGB565Matrix  matrixYUV
     // AVIF_RGB_FORMAT_BGRA      *ToARGBMatrix    matrixYUV
-    // AVIF_RGB_FORMAT_RGBA      *ToARGBMatrix    matrixYVU
     // AVIF_RGB_FORMAT_ABGR      *ToRGBAMatrix    matrixYUV
-    // AVIF_RGB_FORMAT_ARGB      *ToRGBAMatrix    matrixYVU
-
-    if (rgb->format == AVIF_RGB_FORMAT_RGB) {
-        // AVIF_RGB_FORMAT_RGB   *ToRGB24Matrix  matrixYVU
-
-        if (image->yuvFormat == AVIF_PIXEL_FORMAT_YUV420) {
-            if (I420ToRGB24Matrix(image->yuvPlanes[AVIF_CHAN_Y],
-                                  image->yuvRowBytes[AVIF_CHAN_Y],
-                                  image->yuvPlanes[AVIF_CHAN_V],
-                                  image->yuvRowBytes[AVIF_CHAN_V],
-                                  image->yuvPlanes[AVIF_CHAN_U],
-                                  image->yuvRowBytes[AVIF_CHAN_U],
-                                  rgb->pixels,
-                                  rgb->rowBytes,
-                                  matrixYVU,
-                                  image->width,
-                                  image->height) != 0) {
-                return AVIF_RESULT_REFORMAT_FAILED;
-            }
-            return AVIF_RESULT_OK;
-        }
-    } else if (rgb->format == AVIF_RGB_FORMAT_BGR) {
-        // AVIF_RGB_FORMAT_BGR   *ToRGB24Matrix  matrixYUV
+    // AVIF_RGB_FORMAT_RGB_565   *ToRGB565Matrix  matrixYUV
 
-        if (image->yuvFormat == AVIF_PIXEL_FORMAT_YUV420) {
-            if (I420ToRGB24Matrix(image->yuvPlanes[AVIF_CHAN_Y],
+    // Look up table for YUV To RGB Matrix (without filter).
```

Comment:
Would be slightly faster to understand if the `[7][5]` coordinates were described.

Reply:
Good point. It's basically the number of RGB formats and number of YUV formats.

Some projects have a delimiter enum like AVIF_RGB_FORMAT_COUNT or AVIF_RGB_FORMAT_MAX. We don't have that in libavif, so i just used the values here.

I have replaced them with a local #define.